tox.ini

```
@@ -68,7 +68,8 @@ allowlist_externals = bashate
 commands = bashate --verbose {toxinidir}/build.sh {toxinidir}/tools/test/functest.sh
 
 [testenv:functional]
-commands = {toxinidir}/tools/test/functest.sh
+allowlist_externals = bash
```

With `tox-4.4.6-1` locally I need to set

```suggestion
allowlist_externals = {toxinidir}/tools/test/functest.sh
```

Allowing `bash` does not work.

---

Hmm, I'm running tox 4.6.3 and it works, and most of the charm-* repos have bash in allowlist_externals. Did you have the updated 'commands' line?

---

I think either will work, just copied what I saw in neutron and nova.

---

Ah ok. I suppose the advantage of your change is that it is now explicit that the script is really bash.